rlang/pom.xml

```diff
@@ -116,18 +116,10 @@
 
         <dependency>
             <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-client</artifactId>
+            <artifactId>hadoop-client-runtime</artifactId>
             <version>${hadoop.version}</version>
             <scope>compile</scope>
         </dependency>
-
-        <dependency>
-            <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-common</artifactId>
-            <version>${hadoop.version}</version>
-            <scope>compile</scope>
-        </dependency>
-
         <dependency>
```

Comment:
Thanks for checking. I think `hadoop-client-runtime` will pull `hadoop-client-api`, thus there should be no problem.

This PR is not ready (I'm a little busy these days, I may can continue this work next week)

Reply:
`hadoop-client-api` is a runtime dependency of `hadoop-client-runtime`, so `maven-shaded-plugin` should include it. However, I have created the R interpreter package on my local machine and without adding the dependency explicitly, `hadoop-client-api` is not present in the shaded jar. (But maybe I am doing somthing wrong.)

Reply:
How do you know that the maven-shade plugin also adds runtime dependencies? After a first overview I think that the dependency `hadoop-client-api` has to be added manually.

Reply:
The description at https://maven.apache.org/plugins/maven-shade-plugin/shade-mojo.html says:

> Requires dependency resolution of artifacts in scope: runtime

But I am really not sure, how all dependencies are resolved.

Reply:
@matthias-koch the `maven-shaded-plugin` will explicitly print the logs when you perform the `mvn package` command, something like

```
[INFO] --- maven-shade-plugin:3.4.1:shade (default) @ zeppelin-flink-cmd ---
[INFO] Including org.apache.zeppelin:zeppelin-shell:jar:0.11.0-SNAPSHOT in the shaded jar.
[INFO] Including com.google.code.gson:gson:jar:2.8.9 in the shaded jar.
[INFO] Including org.apache.commons:commons-lang3:jar:3.12.0 in the shaded jar.
[INFO] Excluding org.apache.zeppelin:zeppelin-interpreter-shaded:jar:0.11.0-SNAPSHOT from the shaded jar.
...
```

Reply:
I do not see the dependency `hadoop-client-api` in the output.

Reply:
alright, I know where the issue comes from. 

Do we want to respect the `hadoop.deps.scope` if we switch to the Hadoop Shaded client?

Note: Spark drops such profiles and always ships Hadoop Shaded client jars into the binary artifacts.

```
<hadoop.deps.scope>provided</hadoop.deps.scope>
...
<profile>
  <id>include-hadoop</id>
  <properties>
    <hadoop.deps.scope>compile</hadoop.deps.scope>
  </properties>
</profile>
```

Reply:
> As far as I understand it correctly, YARN has already loaded the Hadoop dependencies and therefore they don't need to be in Zeppelin additionally.

There is a switch in YARN to enable/disable Hadoop class population in the containers.

> we should make sure that we do not deliver the library more than once

QQ, I understand we should not include Hadoop classes in plugins, because they will be loaded into the same JVM with Zeppelin server, so that they can share the Hadoop classes. What about the interpreteres? I assume the interpreters are always run in dedicated JVMs, so Hadoop classes seem always necessary (except for those runtimes who already provided Hadoop classes, e.g. Spark, Flink)?

Reply:
I revised this question, the rlang interpreter can work both independently and combine with SparkR, for independent case, Hadoop deps are not required, for SparkR case, the Hadoop jars are handled by Spark runtime, either ship Hadoop jars in SPARK_HOME/jars, or setup HADOOP_CLASSPATH properly, so the rlang interpreter don't need to ship Hadoop deps